lib/ProductOpener/Web.pm

```diff
@@ -311,7 +325,53 @@ sub get_languages_options_list ($target_lc) {
 		);
 	}
 
+	# cache
+	$lang_options_cache{$target_lc} = \@lang_options;
+
 	return \@lang_options;
 }
 
+# cache for get_countries
+my %countries_options_lists = ();
+
+=head2 get_countries_options_list( $target_lc )
+
+Generates all the countries name in the $lc language suitable for an select option list
+
+=head3 Arguments
+
+=head4 $target_lc - language code for labels
+
+=head3 Return value
+
+A reference to a list of hashes with every country code and their label in the $lc language
+[{value => "fr", label => "France"},…]
+
+=cut
+
+sub get_countries_options_list ($target_lc) {
```

Comment:
There's similar code in /cgi/country.pl which is used to display the drop down of countries in the header of each page. In fact we could reuse the dropdown.

Maybe we could merge the two. The one in country.pl is missing caching, that's a good idea.